shared/src/ledger/pos/storage.rs

```diff
@@ -529,6 +598,24 @@ where
             .unwrap();
     }
 
+    fn write_validator_eth_cold_key(
+        &mut self,
+        address: &Self::Address,
+        value: &types::ValidatorEthKey<Self::PublicKey>,
+    ) {
+        self.write(&validator_eth_cold_key_key(address), encode(value))
+            .unwrap();
+    }
+
+    fn write_validator_eth_hot_key(
+        &mut self,
+        address: &Self::Address,
+        value: &types::ValidatorEthKey<Self::PublicKey>,
+    ) {
```

Comment:
yeah this is more of a general issue outside of this PR, idk if @tzemanovic wants to pitch in on this

Reply:
In general, yes agree that it'd be better. We didn't worry about this too much for now in here as this is inside `PosBase` and these methods are only used by the protocol to init chain. If we return an error from, it would still crash further up the stack - the API of PoS crate could do some bigger refactors in general, but for now it's not a high priority